### PR TITLE
Remove Taxonomy.required, make allow_multiple True by default [FC-0030]

### PR DESCRIPTION
**Pull request by bradenmacdonald**
Part of https://github.com/openedx/modular-learning/issues/85 , follows https://github.com/openedx/openedx-learning/pull/87 .

This removes the `Taxonomy.required` field which we don't really need and isn't in the UX designs.

It also makes `allow_multiple` True by default since all user-created taxonomies will need to allow multiple tags and for the MVP we don't plan on implementing any way for users to change Taxonomies to not allow multiple tags.

It also **removes the language tags fixture** because language tags are now auto-created as needed when the system goes to tag something by language. And because removing the `required` field made the fixture stop working in its migration.

There is a corresponding edx-platform PR as well

Private-ref: [FAL-3477](https://tasks.opencraft.com/browse/FAL-3477)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @bradenmacdonald!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: openedx_tagging/core/tagging/api.py [hunk 1]
```diff
@@ -26,7 +26,6 @@ def create_taxonomy(
     name: str,
     description: str | None = None,
     enabled=True,
-    required=False,
     allow_multiple=False,
```
Comment (pomegranited) [inline openedx_tagging/core/tagging/api.py:29]: Could you change this to match the new database default?

```suggestion
    allow_multiple=True,
```

File: openedx_tagging/core/tagging/rest_api/v1/views.py [hunk 2]
```diff
@@ -78,8 +78,6 @@ class TaxonomyView(ModelViewSet):
           applying tags from this taxonomy to an object.
         * enabled (optional): Only enabled taxonomies will be shown to authors
           (default: true).
-        * required (optional): Indicates that one or more tags from this
-          taxonomy must be added to an object (default: False).
         * allow_multiple (optional): Indicates that multiple tags from this
           taxonomy may be added to an object (default: False).
```
Comment (pomegranited) [inline openedx_tagging/core/tagging/rest_api/v1/views.py:82]: nit:
```suggestion
          taxonomy may be added to an object (default: True).
```

**Review by pomegranited — APPROVED** [review]
I left a couple of minor nits, but this is good to merge once they're addressed.

Thanks @bradenmacdonald !

:+1:

- [x] I tested this by installing this branch on my devstack, and ensuring that the added migrations update ok from `main`. 
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] Commit structure follows [OEP-0051](https://github.com/edx/open-edx-proposals/pull/189)

**Review by ChrisChV — APPROVED** [review]

**Review by ChrisChV — APPROVED** [review]
@bradenmacdonald Looks good :+1:  Thanks!

**Comment by bradenmacdonald** [comment]
Thanks for the reviews!

**Comment by openedx-webhooks** [comment]
@bradenmacdonald 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/91&entry.1484655318=https://github.com/bradenmacdonald&entry.752974735=2023-10-04+18:18&entry.1917517419=2023-10-06+02:49&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**